src/routes.js

```diff
@@ -44,4 +45,24 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:username', async ctx => {
+  var user = await User.findOne({ username: ctx.params.username });
+  if (!user) {
+    throw new NotFound();
+  }
+  if (user.id !== ctx.state.currentUser.id) {
+    ctx.request.flash('error', 'You\'re not allowed to edit someone else\'s profile');
+    return ctx.redirect('back');
+  }
+  var { bio } = ctx.request.body;
+  if (isBlank(bio)) {
+    ctx.request.flash('error', 'Your bio looks empty!');
+    return ctx.redirect('back');
+  }
+  var id = user.id;
+  clean(bio);
```

Comment:
Two things: first, this is a no-op (doing things to a string always returns a new string). Also, you should do this _before_ you do your `isBlank` check. Otherwise, someone could put in a single space without triggering your warning, yet in the database the bio will be blank.

Rule of thumb: always perform data "fixing" _before_ you do validations. That way, data that is in the database will _always_ be valid. In theory, if you were to run "validations" on your database data, it should all pass.